backend/entities/room_entity.py

```diff
@@ -27,6 +27,10 @@ class RoomEntity(EntityBase):
     nickname: Mapped[str] = mapped_column(String)
     reservable: Mapped[bool] = mapped_column(Boolean)
 
+    # Optional Office Hours Section ID
+    # NOTE: If null, indicates "official" room; otherwise, indicates entity is created for office hours use.
+    section_id: Mapped[int | None] = mapped_column(Integer, nullable=True)
```

Comment:
Is this field needed here since there is already a relationship defined in the `section_entity`? My concern is that many sections might use one room, so having a section ID here might not be able to store all that data! If this is only used to determine whether a room is used for office hours or not, making this a boolean field would be fine.

Reply:
`section_id` here is more just an indicator of whether it's an official room rather than establishing a relationship.

This would be relevant if a UTA/GTA/Instructor wanted to create an OH event and the location they wanted to host the event doesn't exist in the CSXL room db. Then, they can create a custom, "unofficial" room, at which a `section_id` would be passed too. We didn't implement this functionality in this iteration, so the field could be taken out. Let me know your thoughts!

Reply:
@meghansun322 hmm, in the case where an instructor wants to host OH in a room that does not yet exist, there might be a way to create the room entity, and then map a section to it! The concern here is that `section_id` is not a foreign key, so it is not necessarily being related to section in particular. If there are multiple sections for a room in the future, or if a section moves rooms, this ID will then become outdated. In the case that one room is also an official office hours room for two courses at once (currently the situation for COMP 423 and COMP 311), there could be issues.

If this field was a boolean similar to `reservable` but for OH instead (like maybe `oh_room: bool`), then it can serve as a flag for if the room is an official office hours room or not!

Reply:
Makes sense! Since we didn't implement the feature yet for a staff member to create a custom location for their OH, we're going to delete this field and wait until that feature is added to make that decision. Thanks!